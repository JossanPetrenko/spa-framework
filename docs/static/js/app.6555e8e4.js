(window.webpackJsonp=window.webpackJsonp||[]).push([[1],{"./.docz/app/db.json":function(e){e.exports={config:{title:"Nimbus",description:"Nimbus - SPA Framework",themeConfig:{},menu:[],ordering:"descending",repository:null,native:!1,plugins:[{},{}],indexHtml:".docz/index_template.html",base:"/spa-framework/"},entries:{"packages/@nimbus/styles/readme.mdx":{name:"nimbus/styles",menu:"Api",id:"6293ed660430b98f5c0ad37f63823705",filepath:"packages/@nimbus/styles/readme.mdx",link:null,slug:"packages-nimbus-styles-readme",route:"/packages-nimbus-styles-readme",order:0,headings:[{depth:2,slug:"nimbusstyles",value:"@nimbus /styles"}]},"packages/@nimbus/button/docs/button.mdx":{name:"nimbus/button",menu:"Components",id:"94c0711901e304469ede6258c2b3c0f8",filepath:"packages/@nimbus/button/docs/button.mdx",link:null,slug:"packages-nimbus-button-docs-button",route:"/packages-nimbus-button-docs-button",order:0,headings:[{depth:1,slug:"button",value:"Button"},{depth:2,slug:"padr\xe3o",value:"Padr\xe3o"},{depth:3,slug:"exemplos",value:"Exemplos"},{depth:2,slug:"enfatizado",value:"Enfatizado"},{depth:3,slug:"exemplos-1",value:"Exemplos"},{depth:2,slug:"inten\xe7\xf5es",value:"Inten\xe7\xf5es"},{depth:3,slug:"exemplos-2",value:"Exemplos"},{depth:2,slug:"varia\xe7\xf5es",value:"Varia\xe7\xf5es"},{depth:3,slug:"exemplos---outlined",value:"Exemplos - Outlined"},{depth:3,slug:"exemplos---flat",value:"Exemplos - Flat"},{depth:2,slug:"\xcdcone",value:"\xcdcone"},{depth:3,slug:"exemplos-font-awesome-icons--material-desing-icons",value:"Exemplos (Font Awesome Icons / Material Desing Icons)"}]},"packages/@nimbus/entry/docs/entry.mdx":{name:"nimbus/entry",menu:"Components",id:"8c029e5f77ac97be9736b7b1e39ccc8b",filepath:"packages/@nimbus/entry/docs/entry.mdx",link:null,slug:"packages-nimbus-entry-docs-entry",route:"/packages-nimbus-entry-docs-entry",order:0,headings:[{depth:1,slug:"entry",value:"Entry"},{depth:2,slug:"text-entry",value:"Text Entry"},{depth:3,slug:"exemplo-simples",value:"Exemplo Simples"},{depth:3,slug:"exemplos-label",value:"Exemplos Label"},{depth:3,slug:"exemplos-disabled",value:"Exemplos Disabled"},{depth:3,slug:"exemplos-width",value:"Exemplos Width"},{depth:3,slug:"exemplos-icon",value:"Exemplos Icon"}]},"packages/@nimbus/entry/docs/label.mdx":{name:"nimbus/label",menu:"Components",id:"27d971dbc0734c71e55bad54fb466f07",filepath:"packages/@nimbus/entry/docs/label.mdx",link:null,slug:"packages-nimbus-entry-docs-label",route:"/packages-nimbus-entry-docs-label",order:0,headings:[{depth:1,slug:"entry-label",value:"Entry Label"},{depth:2,slug:"label",value:"Label"},{depth:3,slug:"exemplos",value:"Exemplos"}]},"packages/@nimbus/i18n/docs/i18n.mdx":{name:"nimbus/i18n",menu:"Api",id:"b659672b6ce7352e96793a2eee0efabf",filepath:"packages/@nimbus/i18n/docs/i18n.mdx",link:null,slug:"packages-nimbus-i18n-docs-i18n",route:"/packages-nimbus-i18n-docs-i18n",order:0,headings:[{depth:1,slug:"internationalization",value:"Internationalization"},{depth:2,slug:"uso-b\xe1sico",value:"Uso B\xe1sico"},{depth:2,slug:"v\xe1riaveis",value:"V\xe1riaveis"},{depth:2,slug:"parametros",value:"Parametros"},{depth:2,slug:"contagem-inteligente",value:"Contagem Inteligente"},{depth:2,slug:"mudan\xe7a-de-linguagem",value:"Mudan\xe7a de linguagem"},{depth:2,slug:"exemplo-no-react",value:"Exemplo no React"}]},"packages/@nimbus/layout/docs/sidemenu.mdx":{name:"nimbus/layout",menu:"Components",id:"ffb421ca5e14eeadc70208b9bafbc3bb",filepath:"packages/@nimbus/layout/docs/sidemenu.mdx",link:null,slug:"packages-nimbus-layout-docs-sidemenu",route:"/packages-nimbus-layout-docs-sidemenu",order:0,headings:[{depth:1,slug:"layout",value:"Layout"}]}}}},"./.docz/app/imports.js":function(e,n,s){"use strict";s.d(n,"a",function(){return a});var a={"packages/@nimbus/styles/readme.mdx":function(){return s.e(2).then(s.bind(null,"./packages/@nimbus/styles/readme.mdx"))},"packages/@nimbus/button/docs/button.mdx":function(){return Promise.all([s.e(0),s.e(3)]).then(s.bind(null,"./packages/@nimbus/button/docs/button.mdx"))},"packages/@nimbus/entry/docs/entry.mdx":function(){return Promise.all([s.e(0),s.e(4)]).then(s.bind(null,"./packages/@nimbus/entry/docs/entry.mdx"))},"packages/@nimbus/entry/docs/label.mdx":function(){return Promise.all([s.e(0),s.e(5)]).then(s.bind(null,"./packages/@nimbus/entry/docs/label.mdx"))},"packages/@nimbus/i18n/docs/i18n.mdx":function(){return s.e(6).then(s.bind(null,"./packages/@nimbus/i18n/docs/i18n.mdx"))},"packages/@nimbus/layout/docs/sidemenu.mdx":function(){return Promise.all([s.e(0),s.e(7)]).then(s.bind(null,"./packages/@nimbus/layout/docs/sidemenu.mdx"))}}},"./.docz/app/index.jsx":function(e,n,s){"use strict";s.r(n);var a=s("./node_modules/react/index.js"),t=s.n(a),u=s("./node_modules/react-dom/index.js"),l=s.n(u),o=s("./.docz/app/root.jsx"),d=[],i=[],m=function(){return i.forEach(function(e){return e&&e()})},c=document.querySelector("#root");!function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:o.a;d.forEach(function(e){return e&&e()}),l.a.render(t.a.createElement(e,null),c,m)}(o.a)},"./.docz/app/root.jsx":function(e,n,s){"use strict";(function(e){var a=s("./node_modules/react/index.js"),t=s.n(a),u=s("./node_modules/react-hot-loader/index.js"),l=s("./node_modules/docz-theme-default/dist/index.js"),o=s.n(l),d=s("./.docz/app/imports.js"),i=s("./.docz/app/db.json"),m=function(){return t.a.createElement(o.a,{db:i,imports:d.a,hashRouter:!1})};n.a=Object(u.hot)(e)(m),m.__docgenInfo={description:"",methods:[],displayName:"Root"}}).call(this,s("./node_modules/webpack/buildin/harmony-module.js")(e))},0:function(e,n,s){s("./node_modules/@babel/polyfill/lib/index.js"),e.exports=s("./.docz/app/index.jsx")},1:function(e,n){},2:function(e,n){},3:function(e,n){},4:function(e,n){}},[[0,8,0]],[0,3,4,5,7,6,2]]);