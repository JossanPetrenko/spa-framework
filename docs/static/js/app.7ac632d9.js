(window.webpackJsonp=window.webpackJsonp||[]).push([[1],{"./.docz/app/db.json":function(e){e.exports={config:{title:"Nimbus",description:"Nimbus - SPA Framework",themeConfig:{},menu:[],ordering:"descending",repository:null,native:!1,plugins:[{},{}],indexHtml:".docz/index_template.html"},entries:{"packages/@nimbus/button/button.mdx":{name:"nimbus/button",menu:"Components",id:"6f2239e3c6c70200bb02732d77430039",filepath:"packages/@nimbus/button/button.mdx",link:null,slug:"packages-nimbus-button-button",route:"/packages-nimbus-button-button",order:0,headings:[{depth:1,slug:"button",value:"Button"},{depth:2,slug:"padr\xe3o",value:"Padr\xe3o"},{depth:3,slug:"exemplos",value:"Exemplos"},{depth:2,slug:"enfatizado",value:"Enfatizado"},{depth:3,slug:"exemplos-1",value:"Exemplos"},{depth:2,slug:"inten\xe7\xf5es",value:"Inten\xe7\xf5es"},{depth:3,slug:"exemplos-2",value:"Exemplos"},{depth:2,slug:"varia\xe7\xf5es",value:"Varia\xe7\xf5es"},{depth:3,slug:"exemplos---outlined",value:"Exemplos - Outlined"},{depth:3,slug:"exemplos---flat",value:"Exemplos - Flat"},{depth:2,slug:"\xcdcone",value:"\xcdcone"},{depth:3,slug:"exemplos-font-awesome-icons--material-desing-icons",value:"Exemplos (Font Awesome Icons / Material Desing Icons)"}]},"packages/@nimbus/entry/entry.mdx":{name:"nimbus/entry",menu:"Components",id:"876a39f3bf5d2b6988e27feef58209df",filepath:"packages/@nimbus/entry/entry.mdx",link:null,slug:"packages-nimbus-entry-entry",route:"/packages-nimbus-entry-entry",order:0,headings:[{depth:1,slug:"entry",value:"Entry"},{depth:2,slug:"text-entry",value:"Text Entry"},{depth:3,slug:"exemplo-simples",value:"Exemplo Simples"},{depth:3,slug:"exemplos-label",value:"Exemplos Label"},{depth:3,slug:"exemplos-disabled",value:"Exemplos Disabled"},{depth:3,slug:"exemplos-width",value:"Exemplos Width"},{depth:3,slug:"exemplos-icon",value:"Exemplos Icon"}]},"packages/@nimbus/entry/label.mdx":{name:"nimbus/label",menu:"Components",id:"35b18823f39ebeb545340fb9da304f91",filepath:"packages/@nimbus/entry/label.mdx",link:null,slug:"packages-nimbus-entry-label",route:"/packages-nimbus-entry-label",order:0,headings:[{depth:1,slug:"entry-label",value:"Entry Label"},{depth:2,slug:"label",value:"Label"},{depth:3,slug:"exemplos",value:"Exemplos"}]},"packages/@nimbus/i18n/i18n.mdx":{name:"nimbus/i18n",menu:"Api",id:"57c1624df491bb28d8c6b12b20cd6bd3",filepath:"packages/@nimbus/i18n/i18n.mdx",link:null,slug:"packages-nimbus-i18n-i18n",route:"/packages-nimbus-i18n-i18n",order:0,headings:[{depth:1,slug:"internationalization",value:"Internationalization"},{depth:2,slug:"uso-b\xe1sico",value:"Uso B\xe1sico"},{depth:2,slug:"v\xe1riaveis",value:"V\xe1riaveis"},{depth:2,slug:"parametros",value:"Parametros"},{depth:2,slug:"contagem-inteligente",value:"Contagem Inteligente"},{depth:2,slug:"mudan\xe7a-de-linguagem",value:"Mudan\xe7a de linguagem"},{depth:2,slug:"exemplo-no-react",value:"Exemplo no React"}]},"packages/@nimbus/styles/readme.mdx":{name:"nimbus/styles",menu:"Api",id:"6293ed660430b98f5c0ad37f63823705",filepath:"packages/@nimbus/styles/readme.mdx",link:null,slug:"packages-nimbus-styles-readme",route:"/packages-nimbus-styles-readme",order:0,headings:[{depth:2,slug:"nimbusstyles",value:"@nimbus /styles"}]}}}},"./.docz/app/imports.js":function(e,n,s){"use strict";s.d(n,"a",function(){return t});var t={"packages/@nimbus/button/button.mdx":function(){return Promise.all([s.e(0),s.e(2)]).then(s.bind(null,"./packages/@nimbus/button/button.mdx"))},"packages/@nimbus/entry/entry.mdx":function(){return Promise.all([s.e(0),s.e(3)]).then(s.bind(null,"./packages/@nimbus/entry/entry.mdx"))},"packages/@nimbus/entry/label.mdx":function(){return Promise.all([s.e(0),s.e(4)]).then(s.bind(null,"./packages/@nimbus/entry/label.mdx"))},"packages/@nimbus/i18n/i18n.mdx":function(){return s.e(5).then(s.bind(null,"./packages/@nimbus/i18n/i18n.mdx"))},"packages/@nimbus/styles/readme.mdx":function(){return s.e(6).then(s.bind(null,"./packages/@nimbus/styles/readme.mdx"))}}},"./.docz/app/index.jsx":function(e,n,s){"use strict";s.r(n);var t=s("./node_modules/react/index.js"),a=s.n(t),l=s("./node_modules/react-dom/index.js"),u=s.n(l),o=s("./.docz/app/root.jsx"),i=[],d=[],m=function(){return d.forEach(function(e){return e&&e()})},p=document.querySelector("#root");!function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:o.a;i.forEach(function(e){return e&&e()}),u.a.render(a.a.createElement(e,null),p,m)}(o.a)},"./.docz/app/root.jsx":function(e,n,s){"use strict";(function(e){var t=s("./node_modules/react/index.js"),a=s.n(t),l=s("./node_modules/react-hot-loader/index.js"),u=s("./node_modules/docz-theme-default/dist/index.js"),o=s.n(u),i=s("./.docz/app/imports.js"),d=s("./.docz/app/db.json"),m=function(){return a.a.createElement(o.a,{db:d,imports:i.a,hashRouter:!1})};n.a=Object(l.hot)(e)(m),m.__docgenInfo={description:"",methods:[],displayName:"Root"}}).call(this,s("./node_modules/docz/node_modules/webpack/buildin/harmony-module.js")(e))},0:function(e,n,s){s("./node_modules/@babel/polyfill/lib/index.js"),e.exports=s("./.docz/app/index.jsx")},1:function(e,n){},2:function(e,n){},3:function(e,n){},4:function(e,n){}},[[0,7,0]],[0,2,3,4,5,6]]);