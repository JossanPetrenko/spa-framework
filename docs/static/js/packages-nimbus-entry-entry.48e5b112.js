(window.webpackJsonp=window.webpackJsonp||[]).push([[3],{"./packages/@nimbus/entry/entry.mdx":function(e,o,l){"use strict";l.r(o);var a=l("./node_modules/react/index.js"),n=l.n(a),c=l("./node_modules/@mdx-js/tag/dist/index.js"),t=(l("./packages/@nimbus/styles/scss/theme.scss"),l("./packages/@nimbus/strings/index.js")),m=l("./node_modules/react-grid-system/build/index.js"),d=l("./node_modules/docz/dist/index.m.js"),r=l("./packages/@nimbus/entry/index.js");function p(e,o){if(null==e)return{};var l,a,n=function(e,o){if(null==e)return{};var l,a,n={},c=Object.keys(e);for(a=0;a<c.length;a++)l=c[a],o.indexOf(l)>=0||(n[l]=e[l]);return n}(e,o);if(Object.getOwnPropertySymbols){var c=Object.getOwnPropertySymbols(e);for(a=0;a<c.length;a++)l=c[a],o.indexOf(l)>=0||Object.prototype.propertyIsEnumerable.call(e,l)&&(n[l]=e[l])}return n}o.default=function(e){var o=e.components,l=p(e,["components"]);return n.a.createElement(c.MDXTag,{name:"wrapper",components:o},n.a.createElement(c.MDXTag,{name:"h1",components:o,props:{id:"entry"}},n.a.createElement(c.MDXTag,{name:"strong",components:o,parentName:"h1"},"Entry")),n.a.createElement(d.PropsTable,{of:r.b}),n.a.createElement(c.MDXTag,{name:"h2",components:o,props:{id:"text-entry"}},"Text Entry"),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplo-simples"}},"Exemplo Simples"),n.a.createElement(d.Playground,{__position:1,__codesandbox:"undefined",__code:"<Text label={t('common.label')} placeholder={t('common.placeholder')} />",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder")})),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-label"}},"Exemplos Label"),n.a.createElement(d.Playground,{__position:2,__codesandbox:"undefined",__code:"<Row>\n  <Col md={4}>\n    <Text label={t('common.label')} placeholder={t('common.placeholder')} />\n  </Col>\n  <Col md={4}>\n    <Text label placeholder={t('common.placeholder')} />\n  </Col>\n\n  <Col md={4}>\n    <Text placeholder={t('common.placeholder')} />\n  </Col>\n</Row>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(m.Row,null,n.a.createElement(m.Col,{md:4},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder")})),n.a.createElement(m.Col,{md:4},n.a.createElement(r.b,{label:!0,placeholder:Object(t.a)("common.placeholder")})),n.a.createElement(m.Col,{md:4},n.a.createElement(r.b,{placeholder:Object(t.a)("common.placeholder")})))),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-disabled"}},"Exemplos Disabled"),n.a.createElement(d.Playground,{__position:3,__codesandbox:"undefined",__code:"<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  disabled\n/>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),disabled:!0})),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-width"}},"Exemplos Width"),n.a.createElement(d.Playground,{__position:4,__codesandbox:"undefined",__code:"<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  width=\"50%\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  width=\"392px\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  width=\"49rem\"\n/>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),width:"50%"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),width:"392px"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),width:"49rem"})),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-icon"}},"Exemplos Icon"),n.a.createElement(d.Playground,{__position:5,__codesandbox:"undefined",__code:"<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  iconLeft=\"fas fa-cloud\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  iconRight=\"fas fa-cloud\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  iconRight=\"fas fa-cloud\"\n  iconLeft=\"fas fa-cloud\"\n/>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),iconLeft:"fas fa-cloud"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),iconRight:"fas fa-cloud"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),iconRight:"fas fa-cloud",iconLeft:"fas fa-cloud"})))}}}]);