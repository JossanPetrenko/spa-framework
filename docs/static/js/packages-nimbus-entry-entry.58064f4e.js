(window.webpackJsonp=window.webpackJsonp||[]).push([[3],{"./packages/@nimbus/entry/entry.mdx":function(e,n,o){"use strict";o.r(n);var t=o("./node_modules/react/index.js"),l=o.n(t),a=o("./node_modules/@mdx-js/tag/dist/index.js"),c=(o("./packages/@nimbus/styles/scss/theme.scss"),o("./packages/@nimbus/i18n/index.js")),r=o("./node_modules/react-grid-system/build/index.js"),m=o("./node_modules/docz/dist/index.m.js"),d=o("./packages/@nimbus/entry/index.js");function u(){var e=M(["common.placeholder"]);return u=function(){return e},e}function p(){var e=M(["common.label"]);return p=function(){return e},e}function i(){var e=M(["common.placeholder"]);return i=function(){return e},e}function s(){var e=M(["common.label"]);return s=function(){return e},e}function b(){var e=M(["common.placeholder"]);return b=function(){return e},e}function f(){var e=M(["common.label"]);return f=function(){return e},e}function h(){var e=M(["common.placeholder"]);return h=function(){return e},e}function x(){var e=M(["common.label"]);return x=function(){return e},e}function _(){var e=M(["common.placeholder"]);return _=function(){return e},e}function E(){var e=M(["common.label"]);return E=function(){return e},e}function j(){var e=M(["common.placeholder"]);return j=function(){return e},e}function O(){var e=M(["common.label"]);return O=function(){return e},e}function w(){var e=M(["common.placeholder"]);return w=function(){return e},e}function g(){var e=M(["common.label"]);return g=function(){return e},e}function T(){var e=M(["common.placeholder"]);return T=function(){return e},e}function v(){var e=M(["common.placeholder"]);return v=function(){return e},e}function y(){var e=M(["common.placeholder"]);return y=function(){return e},e}function C(){var e=M(["common.label"]);return C=function(){return e},e}function R(){var e=M(["common.placeholder"]);return R=function(){return e},e}function D(){var e=M(["common.label"]);return D=function(){return e},e}function M(e,n){return n||(n=e.slice(0)),Object.freeze(Object.defineProperties(e,{raw:{value:Object.freeze(n)}}))}function P(e,n){if(null==e)return{};var o,t,l=function(e,n){if(null==e)return{};var o,t,l={},a=Object.keys(e);for(t=0;t<a.length;t++)o=a[t],n.indexOf(o)>=0||(l[o]=e[o]);return l}(e,n);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(t=0;t<a.length;t++)o=a[t],n.indexOf(o)>=0||Object.prototype.propertyIsEnumerable.call(e,o)&&(l[o]=e[o])}return l}n.default=function(e){var n=e.components,o=P(e,["components"]);return l.a.createElement(a.MDXTag,{name:"wrapper",components:n},l.a.createElement(a.MDXTag,{name:"h1",components:n,props:{id:"entry"}},l.a.createElement(a.MDXTag,{name:"strong",components:n,parentName:"h1"},"Entry")),l.a.createElement(m.PropsTable,{of:d.b}),l.a.createElement(a.MDXTag,{name:"h2",components:n,props:{id:"text-entry"}},"Text Entry"),l.a.createElement(a.MDXTag,{name:"h3",components:n,props:{id:"exemplo-simples"}},"Exemplo Simples"),l.a.createElement(m.Playground,{__position:1,__codesandbox:"undefined",__code:"<Text label={t`common.label`} placeholder={t`common.placeholder`} />",__scope:{props:o,t:c.a,Row:r.Row,Col:r.Col,Text:d.b}},l.a.createElement(d.b,{label:Object(c.a)(D()),placeholder:Object(c.a)(R())})),l.a.createElement(a.MDXTag,{name:"h3",components:n,props:{id:"exemplos-label"}},"Exemplos Label"),l.a.createElement(m.Playground,{__position:2,__codesandbox:"undefined",__code:"<Row>\n  <Col md={4}>\n    <Text label={t`common.label`} placeholder={t`common.placeholder`} />\n  </Col>\n  <Col md={4}>\n    <Text label placeholder={t`common.placeholder`} />\n  </Col>\n\n  <Col md={4}>\n    <Text placeholder={t`common.placeholder`} />\n  </Col>\n</Row>",__scope:{props:o,t:c.a,Row:r.Row,Col:r.Col,Text:d.b}},l.a.createElement(r.Row,null,l.a.createElement(r.Col,{md:4},l.a.createElement(d.b,{label:Object(c.a)(C()),placeholder:Object(c.a)(y())})),l.a.createElement(r.Col,{md:4},l.a.createElement(d.b,{label:!0,placeholder:Object(c.a)(v())})),l.a.createElement(r.Col,{md:4},l.a.createElement(d.b,{placeholder:Object(c.a)(T())})))),l.a.createElement(a.MDXTag,{name:"h3",components:n,props:{id:"exemplos-disabled"}},"Exemplos Disabled"),l.a.createElement(m.Playground,{__position:3,__codesandbox:"undefined",__code:"<Text label={t`common.label`} placeholder={t`common.placeholder`} disabled />",__scope:{props:o,t:c.a,Row:r.Row,Col:r.Col,Text:d.b}},l.a.createElement(d.b,{label:Object(c.a)(g()),placeholder:Object(c.a)(w()),disabled:!0})),l.a.createElement(a.MDXTag,{name:"h3",components:n,props:{id:"exemplos-width"}},"Exemplos Width"),l.a.createElement(m.Playground,{__position:4,__codesandbox:"undefined",__code:'<Text\n  label={t`common.label`}\n  placeholder={t`common.placeholder`}\n  width="50%"\n/>\n<Text\n  label={t`common.label`}\n  placeholder={t`common.placeholder`}\n  width="392px"\n/>\n<Text\n  label={t`common.label`}\n  placeholder={t`common.placeholder`}\n  width="49rem"\n/>',__scope:{props:o,t:c.a,Row:r.Row,Col:r.Col,Text:d.b}},l.a.createElement(d.b,{label:Object(c.a)(O()),placeholder:Object(c.a)(j()),width:"50%"}),l.a.createElement(d.b,{label:Object(c.a)(E()),placeholder:Object(c.a)(_()),width:"392px"}),l.a.createElement(d.b,{label:Object(c.a)(x()),placeholder:Object(c.a)(h()),width:"49rem"})),l.a.createElement(a.MDXTag,{name:"h3",components:n,props:{id:"exemplos-icon"}},"Exemplos Icon"),l.a.createElement(m.Playground,{__position:5,__codesandbox:"undefined",__code:'<Text\n  label={t`common.label`}\n  placeholder={t`common.placeholder`}\n  iconLeft="fas fa-cloud"\n/>\n<Text\n  label={t`common.label`}\n  placeholder={t`common.placeholder`}\n  iconRight="fas fa-cloud"\n/>\n<Text\n  label={t`common.label`}\n  placeholder={t`common.placeholder`}\n  iconRight="fas fa-cloud"\n  iconLeft="fas fa-cloud"\n/>',__scope:{props:o,t:c.a,Row:r.Row,Col:r.Col,Text:d.b}},l.a.createElement(d.b,{label:Object(c.a)(f()),placeholder:Object(c.a)(b()),iconLeft:"fas fa-cloud"}),l.a.createElement(d.b,{label:Object(c.a)(s()),placeholder:Object(c.a)(i()),iconRight:"fas fa-cloud"}),l.a.createElement(d.b,{label:Object(c.a)(p()),placeholder:Object(c.a)(u()),iconRight:"fas fa-cloud",iconLeft:"fas fa-cloud"})))}}}]);