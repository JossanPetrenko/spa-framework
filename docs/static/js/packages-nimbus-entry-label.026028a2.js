(window.webpackJsonp=window.webpackJsonp||[]).push([[4],{"./packages/@nimbus/entry/label.mdx":function(e,n,t){"use strict";t.r(n);var a=t("./node_modules/react/index.js"),o=t.n(a),r=t("./node_modules/@mdx-js/tag/dist/index.js"),l=(t("./packages/@nimbus/styles/scss/theme.scss"),t("./packages/@nimbus/i18n/index.js")),s=t("./node_modules/docz/dist/index.m.js"),c=t("./packages/@nimbus/entry/index.js");function m(){var e=i(["common.label"]);return m=function(){return e},e}function p(){var e=i(["common.label"]);return p=function(){return e},e}function i(e,n){return n||(n=e.slice(0)),Object.freeze(Object.defineProperties(e,{raw:{value:Object.freeze(n)}}))}function u(e,n){if(null==e)return{};var t,a,o=function(e,n){if(null==e)return{};var t,a,o={},r=Object.keys(e);for(a=0;a<r.length;a++)t=r[a],n.indexOf(t)>=0||(o[t]=e[t]);return o}(e,n);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)t=r[a],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(o[t]=e[t])}return o}n.default=function(e){var n=e.components,t=u(e,["components"]);return o.a.createElement(r.MDXTag,{name:"wrapper",components:n},o.a.createElement(r.MDXTag,{name:"h1",components:n,props:{id:"entry-label"}},o.a.createElement(r.MDXTag,{name:"strong",components:n,parentName:"h1"},"Entry Label")),o.a.createElement(s.PropsTable,{of:c.a}),o.a.createElement(r.MDXTag,{name:"h2",components:n,props:{id:"label"}},"Label"),o.a.createElement(r.MDXTag,{name:"h3",components:n,props:{id:"exemplos"}},"Exemplos"),o.a.createElement(s.Playground,{__position:1,__code:'<Label>{t`common.label`}</Label>\n<Label htmlFor="rotulo">{t`common.label`} </Label>',__scope:{props:t,t:l.a,Label:c.a}},o.a.createElement(c.a,null,Object(l.a)(p())),o.a.createElement(c.a,{htmlFor:"rotulo"},Object(l.a)(m())," ")))}}}]);