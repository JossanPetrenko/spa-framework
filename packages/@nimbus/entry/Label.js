import React from 'react';
import PropTypes from 'prop-types';
import {css} from '@nimbus/core';

export default class Label extends React.PureComponent {
  render () {
    const {children, ...othersProps} = this.props;

    let classList = css ('n-entry-label');

    return (
      <label className={classList} {...othersProps}>
        {children}
      </label>
    );
  }
}

Label.propTypes = {
  /** Conteúdo da label */
  children: PropTypes.oneOfType ([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node,
    PropTypes.bool,
  ]),

  /** Rótulo para indicar ID */
  htmlFor: PropTypes.string,
};

Label.defaultProps = {};
