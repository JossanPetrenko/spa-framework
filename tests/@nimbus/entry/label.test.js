import React from 'react';
import {expect} from 'chai';
import {t} from '../../../packages/@nimbus/strings/index.js';

import Label from '../../../packages/@nimbus/entry/Label';

it ('deve renderizar a label com texto ', () => {
  const wrapper = shallow (<Label>{t ('common.titulo')}</Label>);
  expect (wrapper).to.matchSnapshot ();
});



