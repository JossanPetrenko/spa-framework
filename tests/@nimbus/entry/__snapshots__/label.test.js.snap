// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve renderizar a label com texto  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label>
    Título
  </Label>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Título",
      "className": "n-entry-label",
    },
    "ref": null,
    "rendered": "Título",
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Título",
        "className": "n-entry-label",
      },
      "ref": null,
      "rendered": "Título",
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
